server/src/pages/shot/controller.js

```diff
@@ -280,15 +282,28 @@ exports.setTitle = function(title) {
 };
 
 function render() {
+  if (!initialRender) {
+    initialRender = true;
+  }
   page.render(model);
 }
 
 document.addEventListener("addon-present", (e) => {
   if (e.detail) {
     const capabilities = JSON.parse(e.detail);
     if (capabilities["copy-to-clipboard"]) {
-      model.canCopy = true;
-      render();
+      // React throw warnings when the client side content does not match the
+      // server side rendered content. Since updating the model here changes the
+      // rendered content, it's performed after the initial rendering.
+      const delayedRender = () => {
+        if (initialRender) {
+          model.canCopy = true;
+          render();
+          return;
+        }
+        setTimeout(delayedRender, 50);
+      };
+      delayedRender();
```

Comment:
Hmm. It seems funny to use a setTimeout here, but I'm not sure of the best workaround.

What if we set `suppressHydrationWarning`, to signal that React should ignore the gap between server + client, or maybe set `state.isClient = true` inside `componentDidMount`, and checked the `canCopy` value from the model if `isClient`?

Reply:
`isClient` works great.  Thanks!  Also moved the event dispatch to a more sensible place.